app/models/transformation_mapping_item.rb

```diff
@@ -4,4 +4,24 @@ class TransformationMappingItem < ApplicationRecord
   belongs_to :destination, :polymorphic => true
 
   validates :source_id, :uniqueness => {:scope => [:transformation_mapping_id, :source_type]}
+
+  validate :source_cluster,      :if => -> { source.kind_of?(EmsCluster) }
+  validate :destination_cluster, :if => -> { source.kind_of?(EmsCluster) || source.kind_of?(CloudTenant) }
```

Comment:
@thearifismail is the source ever a `CloudTenant`?  I thought that was one of the destination options.

Reply:
No.  You are right `CloudTenant` is only for destination.